_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+permalink: on-code-reviews
+title: 'On Code Reviews'
+tags: [programming]
+---
+
+I wanted to share some insights into how I review code, learning from others,
+helping others, and getting a better overview of the code base which ultimately
+allows me to write better[^1] code. This post is inspired by a recent
+[tweet](https://twitter.com/kristoferbaxter/status/1186290473322741760), a
+similar [blog post](https://techbeacon.com/app-dev-testing/10-commandments-navigating-code-reviews)
+(by [Angie Jones](https://twitter.com/techgirl1908)) and a [talk](https://www.youtube.com/watch?v=Ea8EiIPZvh0)
+on the topic (by [April Wensel](https://twitter.com/aprilwensel)) from 2018's
+[try! Swift NYC](https://www.tryswift.co) (which you should totally attend if
+you get the chance).
+
+- **I point out the good things.** This goes for anything really; not just code
+reviews. If there's something you like or learned, let the author know. I
+believe this will make a big difference on how a review is perceived, and is a
+win-win all around.<br />
+This is what I believe to be the **single most important part of a review**.
+- **I assume best intent.** I've written questionable code before, or have
+overseen mistakes. That happens. Always be kind and approach a review
+positively. 
+- **I read the pull request's description _and_ ticket.** Yes, also the ticket.
+This makes the time of everyone involved a lot more valuable. Seldom does this
+not help me getting a better understanding of the code. And understanding is
+key. If something in the description or ticket is unclear to me, I try to reach
+out the author before continuing a review.
```

Comment:
👌 context is very important when reviewing

```suggestion
out to the author before continuing a review.
```

Reply:
Nice catch!